cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -614,22 +640,42 @@ class SimpleArrayMixinCalculators
 
     A add_simd(A const & other) const
     {
-        return A(*static_cast<A const *>(this)).iadd_simd(other);
+        A const * athis = static_cast<A const *>(this);
+        A ret(athis->shape());
```

Comment:
I think RVO / copy elision works for this.  I am curious whether the older way allowed copy elision:

```cpp
A(*static_cast<A const *>(this)).iadd_simd(other);
```